src/geometry/Types.hh

```diff
@@ -8,23 +8,17 @@
 #pragma once
 
 #include "base/OpaqueId.hh"
+#include "base/Types.hh"
 
 namespace celeritas
 {
 class Geometry;
 //---------------------------------------------------------------------------//
 using VolumeId = OpaqueId<Geometry, unsigned int>;
 
-/*!
- * Whether the particle is inside or outside the problem space.
- *
- * The particle tracking should terminate when the boundary state is "outside".
- */
-enum class Boundary : bool
-{
-    outside = false,
-    inside  = true
-};
+// a small extra push to ensure that track goes beyond boundaries
+constexpr static real_type kgeomTolerance = 1.0e-12; // cm (default length
```

Comment:
- This shouldn't need to be `static` (since it's a constexpr global) 
- The formatting for the name is inconsistent with the code style
- Consider whether the tolerance unit should be multiplied explicitly into the constant (I'm not 100% sure it should, since the tolerance doesn't really signify a physical quantity, just a fudge factor)
- Since it's not a "type", and it's related to operating on the geometry track state, I think a `static CELERITAS_CONSTEXPR_FUNCTION real_type tolerance() { return 1e-12; }` as a class function on the track state might be more appropriate. I think I originally had something similar in that class.

Reply:
Keyword 'static' removed and variable renamed as boundary_push, to make its purpose clear. 
Its value will be subject to tuning, especially in the propagation of charged particles.